File: Trees/LevelOrderTraversal.py
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        queue = [] 
        ans = []
        if root is not None:
            queue.append(root)
        while queue:
            length = len(queue)
            arr = []
            for i in range(length):
                cur = queue.pop(0)
                if cur.left is not None:
                    queue.append(cur.left)
                if cur.right is not None:
                    queue.append(cur.right)
                arr.append(cur.val)
            ans.append(arr)
        return ans

File: Blind75/levelOrderTraversal.cpp


class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> answer;
        queue<TreeNode*> queue; 
        if (!root){
            return answer;
        }
        queue.push(root);
        while (!queue.empty()){  
            vector<int> temp;
            int s = queue.size();
            for (int i = 0; i < s; i++){
                TreeNode* cur = queue.front();
                 if (cur->left != NULL){
                    queue.push(cur->left);
                }
                if (cur->right != NULL){
                    queue.push(cur->right);
                }
                temp.push_back(cur->val);
                queue.pop();
            }
            answer.push_back(temp);  
        }
        return answer;
    }
};
